source/MaterialXGenShader/Nodes/OpenColorIONode.cpp

```diff
@@ -0,0 +1,151 @@
+//
+// Copyright Contributors to the MaterialX Project
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#ifdef MATERIALX_BUILD_OCIO
+
+#include <MaterialXGenShader/Nodes/OpenColorIONode.h>
+#include <MaterialXGenShader/OpenColorIOManagementSystem.h>
+
+#include <MaterialXCore/Interface.h>
+#include <MaterialXGenGlsl/GlslShaderGenerator.h>
+#include <MaterialXGenShader/GenContext.h>
+#include <MaterialXGenShader/Library.h>
+#include <MaterialXGenShader/ShaderNode.h>
+#include <MaterialXGenShader/Shader.h>
+#include <MaterialXGenShader/ShaderStage.h>
+
+#include <OpenColorIO/OpenColorIO.h>
+#include <OpenColorIO/OpenColorTypes.h>
+
+#include <cstring>
+#include <functional>
+#include <memory>
+#include <string>
+
+MATERIALX_NAMESPACE_BEGIN
+
+namespace
+{
+// Internal OCIO strings:
+constexpr const char OCIO_COLOR3[] = "color3";
+
+// Lengths where needed:
+constexpr auto OCIO_COLOR3_LEN = sizeof(OCIO_COLOR3) / sizeof(OCIO_COLOR3[0]);
+
+} // namespace
+
+ShaderNodeImplPtr OpenColorIONode::create()
+{
+    return std::make_shared<OpenColorIONode>();
+}
+
+void OpenColorIONode::initialize(const InterfaceElement& element, GenContext& context)
+{
+    ShaderNodeImpl::initialize(element, context);
+
+    // Single function shared between color3 and color4 nodes, use a custom hash with only the function name.
+    _hash = std::hash<string>{}(getFunctionName());
+}
+
+void OpenColorIONode::emitFunctionDefinition(
+    const ShaderNode& /*node*/,
+    GenContext& context,
+    ShaderStage& stage) const
+{
+    if (stage.getName() == Stage::PIXEL)
+    {
+        auto ocioManager = std::dynamic_pointer_cast<OpenColorIOManagementSystem>(context.getShaderGenerator().getColorManagementSystem());
+
+        auto gpuProcessor = ocioManager->getGpuProcessor(getName());
+        OCIO::GpuShaderDescRcPtr shaderDesc = OCIO::GpuShaderDesc::CreateShaderDesc();
+
+        // TODO: Extend to essl and MDL and possibly SLang.
+        if (context.getShaderGenerator().getTarget() == "genglsl") {
+            shaderDesc->setLanguage(OCIO::GPU_LANGUAGE_GLSL_4_0);
+        } else if (context.getShaderGenerator().getTarget() == "genmsl") {
+            shaderDesc->setLanguage(OCIO::GPU_LANGUAGE_MSL_2_0);
```

Comment:
Would be nice if we could keep this version in sync somehow with 

`const string MslShaderGenerator::VERSION = "2.3";`

from [here](https://github.com/AcademySoftwareFoundation/MaterialX/blob/30f694f42670c4eb5e7386ce931a6e3690bbe2a3/source/MaterialXGenMsl/MslShaderGenerator.cpp#L46)